### PR TITLE
config: deprecate `tikv-client.copr-cache.enable` and invisible some copr-cache configs

### PR DESCRIPTION
**Pull request by lzmhhh123**
Signed-off-by: lzmhhh123 <lzmhhh123@gmail.com>

### What problem does this PR solve?

Problem Summary:

### What is changed and how it works?

How it Works: make `capcity>0` check whether it enabled.

### Related changes

- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- remove "enable" field of coprocessor cache configuration.

### CODE REVIEW

**Review by jackysp — COMMENTED** [review]
You may need to add it to `deprecatedConfig` in config.go.

**Comment by bb7133** [comment]
You can follow this PR https://github.com/pingcap/tidb/pull/18618 to deprecate this configuration.

File: config/config.toml.example [hunk 1]
```diff
@@ -418,20 +418,9 @@ store-liveness-timeout = "5s"
...
 [tikv-client.copr-cache]
-# Whether to enable the copr cache. The copr cache saves the result from TiKV Coprocessor in the memory and
-# reuses the result when corresponding data in TiKV is unchanged, on a region basis.
-enable = true
-
-# The capacity in MB of the cache.
+# The capacity in MB of the cache. Zero means to disable coprocessor cache.
 capacity-mb = 1000.0
 
-# Only cache small requests. Zero means no limits for requests.
```
Comment (jackysp) [inline config/config.toml.example:428]: All of them are deprecated? PTAL @breeswish

File: config/config.toml.example [hunk 2]
```diff
@@ -418,20 +418,9 @@ store-liveness-timeout = "5s"
...
-admission-max-ranges = 500
-# Only cache requests whose result set is small.
-admission-max-result-mb = 10.0
-# Only cache requests takes notable time to process.
-admission-min-process-ms = 5
```
Comment (breeswish) [inline config/config.toml.example:433]: Seems that some users are tend to modify these configurations, do we need to remove them?

Comment (lzmhhh123) [reply config/config.toml.example:433]: It's a request from the PM.

Comment (lzmhhh123) [reply config/config.toml.example:433]: @XuHuaiyu PTAL.

Comment (breeswish) [reply config/config.toml.example:433]: @bb7133 may have impression about the client that uses these configuration values. If we drop them, then we cannot serve this kind of clients any more.

Comment (zz-jason) [reply config/config.toml.example:433]: How about:
1. only expose `capacity-mb` to users by default to increase usability.
2. make other config items configurable, but hidden by default.

Comment (breeswish) [reply config/config.toml.example:433]: > How about:
> 
> 1. only expose `capacity-mb` to users by default to increase usability.
> 2. make other config items configurable, but hidden by default.

Hide configurations looks good to me.

**Review by XuHuaiyu — COMMENTED** [review]
/LGTM

**Review by breeswish — APPROVED** [review]

**Comment by lzmhhh123** [comment]
unit test failed at:

```
[2021-03-10T07:06:00.934Z] FAIL: conn_test.go:740: ConnTestSuite.TestTiFlashFallback
[2021-03-10T07:06:00.934Z] 
[2021-03-10T07:06:00.934Z] conn_test.go:785:
[2021-03-10T07:06:00.934Z]     testFallbackWork(c, tk, cc, "select sum(a) from t")
[2021-03-10T07:06:00.934Z] conn_test.go:817:
[2021-03-10T07:06:00.934Z]     c.Assert(cc.handleQuery(ctx, sql), IsNil)
[2021-03-10T07:06:00.934Z] ... value *errors.withStack = [tikv:9005]Region is unavailable ("[tikv:9005]Region is unavailable")
```

**Comment by mahjonp** [comment]
We (the benchbot) uses the `tikv-client.copr-cache.enable = false` to disable `copr-cache`, and found that the`copr-cache` is used by default this week, it's too terrible and inefficient for me to waste time to investigate this incompatible issue.

**Comment by lzmhhh123** [comment]
> We (the benchbot) uses the `tikv-client.copr-cache.enable = false` to disable `copr-cache`, and found that the`copr-cache` is used by default this week, it's too terrible and inefficient for me to waste time to investigate this incompatible issue.

`copr-cache` is used by default before this PR. So not it doesn't break compatibility.

**Comment by mahjonp** [comment]
> > We (the benchbot) uses the `tikv-client.copr-cache.enable = false` to disable `copr-cache`, and found that the`copr-cache` is used by default this week, it's too terrible and inefficient for me to waste time to investigate this incompatible issue.
> 
> `copr-cache` is used by default before this PR. So not it doesn't break compatibility.

IMO I think compatibility means it won't break a user's current configuration even it's not the default configuration(you know it's rare for a user that he/she doesn't override a default value)

**Pull request merged**